flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/handler/util/GetApiVersionHandler.java

```diff
@@ -40,12 +42,31 @@
         extends AbstractSqlGatewayRestHandler<
                 EmptyRequestBody, GetApiVersionResponseBody, EmptyMessageParameters> {
 
+    private final List<SqlGatewayRestAPIVersion> supportedVersions;
+
     public GetApiVersionHandler(
             SqlGatewayService service,
             Map<String, String> responseHeaders,
             MessageHeaders<EmptyRequestBody, GetApiVersionResponseBody, EmptyMessageParameters>
                     messageHeaders) {
+        this(
+                service,
+                responseHeaders,
+                messageHeaders,
+                Arrays.stream(SqlGatewayRestAPIVersion.values())
+                        .filter(SqlGatewayRestAPIVersion::isStableVersion)
+                        .collect(Collectors.toList()));
```

Comment:
Nit: encapsulate the logic within `SqlGatewayRestAPIVersion` like `getStableVersions` or `getSupportedVersions`?